src/swap-helpers.ts

```diff
@@ -267,3 +278,23 @@ export const isLikeKind = (
   }
   return false
 }
+
+export const getTokenId = (
+  coreWallet: EdgeCurrencyWallet,
+  currencyCode: string
+): string | undefined => {
+  const { allTokens } = coreWallet.currencyConfig
+  return Object.keys(allTokens).find(
+    edgeToken => allTokens[edgeToken].currencyCode === currencyCode
+  )
+}
```

Comment:
We need to explicitly handle the `currencyCode === coreWallet.currencyInfo.currencyCode` case, since otherwise it would be possible to add a custom token named "ETH" on the Ethereum network, and have that break the main coin.

Reply:
done

Reply:
ok. good catch.